src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -283,15 +283,20 @@ address MacroAssembler::target_addr_for_insn(address insn_addr, unsigned insn) {
     return address(uint64_t(Instruction_aarch64::extract(insns[0], 20, 5))
                    + (uint64_t(Instruction_aarch64::extract(insns[1], 20, 5)) << 16)
                    + (uint64_t(Instruction_aarch64::extract(insns[2], 20, 5)) << 32));
-  } else if (Instruction_aarch64::extract(insn, 31, 22) == 0b1011100101 &&
-             Instruction_aarch64::extract(insn, 4, 0) == 0b11111) {
-    return 0;
   } else {
     ShouldNotReachHere();
   }
   return address(((uint64_t)insn_addr + (offset << 2)));
 }
 
+address MacroAssembler::target_addr_for_insn_allow_null_ret(address insn_addr, unsigned insn) {
+  if (Instruction_aarch64::extract(insn, 31, 22) == 0b1011100101 &&
```

Comment:
Consider replacing raw constants check with NativeInstruction::is_ldrw_to_zr function call which performs the same check.

Reply:
Good catch. There are actually two uses of the raw constants in
macroAssembler_aarch64.cpp and I've cleaned up both.

Reply:
This was a nice refactoring.

Reply:
Good. Thank you!

Reply:
Good.